src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php

```diff
@@ -205,6 +205,9 @@ public function theProductShouldAppearInTheShop($productName)
      */
     public function iWantToBrowseProducts()
     {
+        // Timeout because of elasticsearch update timing, use "curl 'localhost:9200/_cat/pending_tasks'" to check instead
+        sleep(1);
```

Comment:
Maybe ``->waitFor()`` method inside ``indexPage`` would help you. You can create something like ``->waitForSomething()`` in ``indexPage``.